### DISCUSSION TITLE
Oauth Authentication - Self Hosted

### DISCUSSION THREAD

**Discussion by EagleEyez1** [vrtmrz/obsidian-livesync#484] [General]
I was finally able to setup the CouchDB in a docker container with Traefik reverse proxy with SSL. Livesync is working great between my Macbook and Android. Will test additional tablets later. I had to use Basic Authentication on my container to get it working otherwise the Fetch would throw an error. Has anyone been able to enable a stronger authentication method or use Ouath or another 2 factor authentication? It would be nice to provide better security from bad actors.

**Comment by jdkruzr** [comment]
hey Eric - thanks for bringing this up. does this mean fly.io is not required at all; i.e. the only thing the plugin requires is CouchDB? also, if you were able to set up Basic Authentication with Traefik, in the same way you would be able to implement other kinds of authentication. I know Nginx for example supports a ton of different authentication plugins.

**Comment by AndrewPro** [comment]
Hello Eric, I am trying to figure out how to install this in the same environment. CouchDB in a docker container with Traefik reverse proxy. If you could point me in the right direction I would be grateful for your help. For example how do I host the Obsidian sync plugin on the server side? In a separate container, etc etc.

For the auth, I use Authentik for my self hosted programs, however I am unsure if it will work with the obsidian extension since there is no user interface to login. I think if you have a secure pw, that's all what matters. Like a 20 digit random generated one should suffice.

**Comment by vrtmrz** [comment]
Sorry I have not been able to participate much in the discussion.
This may be able to be addressed in #588. If you are interested, please have a look at the Issue as well.
CouchDB also allows authentication via JWT, which we would like to implement if possible.
